src/_path.h

```
@@ -939,79 +939,151 @@ void cleanup_path(PathIterator &path,
     }
 }
 
+void quad2cubic(double x0, double y0,
+                double x1, double y1,
+                double x2, double y2,
+                double *outx, double *outy)
+{
+
+    outx[0] = x0 + 2./3. * (x1 - x0);
+    outy[0] = y0 + 2./3. * (y1 - y0);
+    outx[1] = outx[0] + 1./3. * (x2 - x0);
+    outy[1] = outy[0] + 1./3. * (y2 - y0);
+    outx[2] = x2;
+    outy[2] = y2;
+}
+
+char *__append_to_string(char *p, char *buffer, size_t buffersize,
+                         const char *content)
+{
+    for (const char *i = content; *i; ++i) {
+        if (p < buffer || p - buffer >= (int)buffersize) {
+            return p;
```

Shouldn't this raise instead of silently truncating or doing nothing?